docs/index.md

```diff
@@ -0,0 +1,268 @@
+object GetMeAValue(sring url) {
+  if (IsMalformed(url)) {
+    throw new Exception($"The url `{url}` is malformed!");
+  }
+  
+  // do something else to load data from the URL
+  return data;
+}
+
+string RenderHtml(object toRender) {
+  // if toRender can't generate valid HTML, throw Error("invalid HTML");
+  // if it can, theRenderedHTML;
+}
+
+void WriteOutput(string html)
+{
+  // I/O
+}
+
+// somewhere else in the codebase -- throws an exception
+var badUrl = "http:/www.google.com";  // missing a slash
+var response = GetMeAValue(badUrl);  // throws here
+
+// we never get here, but it could throw too
+var htmlForPage = RenderHtml(value);
+
+// so we definitely can't get here safely
+WriteOutput(htmlForPage);
+```
+Notice: there’s no way for the caller to know that the function will throw. Perhaps you’re
+very disciplined and write good docstrings for every function – and moreover, perhaps
+everyone’s editor shows it to them and they pay attention to that briefly-available
+popover. More likely, though, this exception throws at runtime and probably as a result of
+user-entered data – and then you’re chasing down the problem through error logs. More, if
+you do want to account for the reality that any function anywhere in C♯ might
+actually throw, you’re going to write something like this:
+```csharp
+try
+{
+  var badUrl = "http:/www.google.com";  // missing a slash
+  var response = GetMeAValue(badUrl);  // throws here
+
+  // we never get here, but it could throw too
+  var htmlForPage = RenderHtml(value);
+
+  // so we definitely can't get here safely
+  WriteOutput(htmlForPage); 
+}
+catch(Exception exn)
+{
+  HandleErr(exn);
+}
+```
+This is like the first example but even worse for repetition!  Also, C♯ can’t help you
+here! There's no type signatures to say “This throws an exception!”
+
+Instead, we can use a `Result` to get us a container type, much like `Maybe`, to let us
+deal with this scenario. A `Result` is either an Ok wrapping around a value (like Just does)
+or an Err wrapping around some type defining what went wrong (not like Nothing, which has
+no contents).
+```csharp
+Result<Payload, string> GetMeAValue(string url)
+{
+  if (IsMalformed(url)) {
+    return Result<Payload, string>.Err($"The url '{url}' is malformed");
+  }
+  
+  // do something else to load data from the url
+  return Result.Ok(data);
+}
+
+Result<string, string> RenderHtml(string toRender)
+{
+  // if toRender can't generate valid HTML, return Err("invalid HTML");
+  // if it can, return Ok(theRenderedHTML);
+}
+
+void WriteOutput(string html)
+{
+  
+}
+
+// somewhere else in the codebase -- no exception this time!
+var badUrl = "http:/www.google.com";  // missing a slash
+
+// value = Err(The url '${http:/www.google.com}' is malformed)
+var value = GetMeAValue(badUrl);
+
+// htmlForPage = the same error! or, if it was Ok, could be a different
+// `Err` (because of how `andThen` works).
+var htmlForPage = value.And(RenderHtml);
```

Comment:
❌ This one is no longer correct since you updated the functions so this is `AndThen` (`Chain`/`Bind`).

```suggestion
var htmlForPage = value.AndThen(RenderHtml);
```